bundle-workflow/scripts/default/build.sh

```diff
@@ -55,15 +59,32 @@ if [ -z "$VERSION" ]; then
     exit 1
 fi
 
-[[ "$SNAPSHOT" == "true" ]] && VERSION=$VERSION-SNAPSHOT
+[[ "$SNAPSHOT" == "true" && "$DASHBOARDS" == "false" ]] && VERSION=$VERSION-SNAPSHOT
 [ -z "$OUTPUT" ] && OUTPUT=artifacts
 
 mkdir -p $OUTPUT
-./gradlew assemble --no-daemon --refresh-dependencies -DskipTests=true -Dopensearch.version=$VERSION -Dbuild.snapshot=$SNAPSHOT
 
-zipPath=$(find . -path \*build/distributions/*.zip)
-distributions="$(dirname "${zipPath}")"
+if [[ "$DASHBOARDS" == "true" ]]; then
+    mkdir -p $OUTPUT/plugins
+    PLUGIN_NAME=$(basename "$PWD")
+    # TODO: [CLEANUP] Needed OpenSearch Dashboards git repo to build the required modules for plugins
+    # This makes it so there is a dependency on having Dashboards pulled already.
+    cp -r ../$PLUGIN_NAME/ ../OpenSearch-Dashboards/plugins
+    echo "BUILD MODULES FOR $PLUGIN_NAME"
+    (cd ../OpenSearch-Dashboards && yarn osd bootstrap)
+    echo "BUILD RELEASE ZIP FOR $PLUGIN_NAME"
+    (cd ../OpenSearch-Dashboards/plugins/$PLUGIN_NAME && yarn plugin-helpers build)
+    echo "COPY $PLUGIN_NAME.zip"
+    cp -r ../OpenSearch-Dashboards/plugins/$PLUGIN_NAME/build/$PLUGIN_NAME-$VERSION.zip $OUTPUT/plugins/
+else
+    ./gradlew assemble --no-daemon --refresh-dependencies -DskipTests=true -Dopensearch.version=$VERSION -Dbuild.snapshot=$SNAPSHOT
+
+    zipPath=$(find . -path \*build/distributions/*.zip)
+    distributions="$(dirname "${zipPath}")"
+
+    echo "COPY ${distributions}/*.zip"
+    mkdir -p $OUTPUT/plugins
+    cp ${distributions}/*.zip ./$OUTPUT/plugins
+fi
```

Comment:
We should test these scripts as part of our CI workflows.
A follow up for both OpenSearch and Dashboards.

Reply:
Opened up an issue: https://github.com/opensearch-project/opensearch-build/issues/591